index.php

```diff
@@ -1,48 +1,103 @@
-	|  of the License, or (at your option) any later version.                      |
-	|                                                                              |
-	|  This program is distributed in the hope that it will be useful,             |
-	|  but WITHOUT ANY WARRANTY; without even the implied warranty of              |
-	|  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the               |
-	|  GNU General Public License for more details.                                |
-	|                                                                              |
-	|  You should have received a copy of the GNU General Public License           |
-	|  along with this program; if not, write to the Free Software                 |
-	|  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA. |
-	+------------------------------------------------------------------------------+
+	+-----------------------+-----------------------------------------------------------------------+
+	|  Script		| PHProxy   +   SabzProxy						|
+	|  Author		| Abdullah Arif								|
+	|  Modifier		| Forgetful  (Hamid R) + Timo Van Neerden + Amaury Carrade + Machou	|
+	|  Last Modified	| 11 january 2016							|
+	+-----------------------+-----------------------------------------------------------------------+
+	|  This program is free software; you can redistribute it and/or				|
+	|  modify it under the terms of the GNU General Public License					|
+	|  as published by the Free Software Foundation; either version 2				|
+	|  of the License, or (at your option) any later version.					|
+	|												|
+	|  This program is distributed in the hope that it will be useful,				|
+	|  but WITHOUT ANY WARRANTY; without even the implied warranty of				|
+	|  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the				|
+	|  GNU General Public License for more details.							|
+	|												|
+	|  You should have received a copy of the GNU General Public License				|
+	|  along with this program; if not, write to the Free Software					|
+	|  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.			|
+	+-----------------------------------------------------------------------------------------------+
 */
 
+// LANGUAGE
+//
+
+// Please, if you want translate it, add same english & french, Russian > "ru", Spain > "es")
+// Find all identification codes here : http://www.metamodpro.com/browser-language-codes
+$language_acceppted = array(
+				'en',	// English
+				'fr'	// French
+			);
+
+function get_language($sDefault = 'en')
+{
+	if(!empty($_SERVER['HTTP_ACCEPT_LANGUAGE']))
+	{
+		$aBrowserLanguages = explode(',', $_SERVER['HTTP_ACCEPT_LANGUAGE']);
+		foreach($aBrowserLanguages as $sBrowserLanguage)
+		{
+			$sLang = strtolower(substr($sBrowserLanguage, 0, 2));
+
+			if(in_array($sLang, $language_acceppted))
+				return $sLang;
+
+			else
+				return $sDefault;
+		}
+	}
+
+	else
+		return $sDefault;
+}
```

Comment:
This is still a problem, sorry.

Here, if the _first_ language tested is not in the list, the default value will be returned, even if a compatible language is present after.

Fix:
```php
function get_language($sDefault = 'en')
{
	if(!empty($_SERVER['HTTP_ACCEPT_LANGUAGE']))
	{
		$aBrowserLanguages = explode(',', $_SERVER['HTTP_ACCEPT_LANGUAGE']);
		foreach($aBrowserLanguages as $sBrowserLanguage)
		{
			$sLang = strtolower(substr($sBrowserLanguage, 0, 2));

			if(in_array($sLang, $language_acceppted))
				return $sLang;
		}
	}

	return $sDefault;
}
```